bpf/bpf_lxc.c

```diff
@@ -1127,7 +1127,8 @@ ipv4_policy(struct __ctx_buff *ctx, int ifindex, __u32 src_label, __u8 *reason,
 	orig_sip = ip4->saddr;
 
 	l4_off = l3_off + ipv4_hdrlen(ip4);
-	csum_l4_offset_and_flags(tuple.nexthdr, &csum_off);
+	if (!ipv4_is_not_first_fragment(ip4))
```

Comment:
We should try to avoid double-negatives in conditional statements like this for better readability, eg:

```suggestion
	if (ipv4_is_first_fragment(ip4))
```

Reply:
I don't think it's good idea. ipv4_is_first_fragment will only allow first fragment. But here, we need all the other non framgmented packets too. "non-first-fragment" has a specific meaning to denote fragmented packets that is not the first one. It's widely used: [google](https://www.google.com/search?newwindow=1&rlz=1C5CHFA_enUS867US867&sxsrf=ALeKk02Jv7eHZx5tApmHsF5_OgWdL4b_tw%3A1602201167134&ei=T6Z_X_DeB8zc-gTksaOIBQ&q=%22Non-first+fragments%22&oq=%22Non-first+fragments%22&gs_lcp=CgZwc3ktYWIQAzIJCAAQyQMQFhAeMgYIABAWEB46BAgAEEc6BAgjECdQj29YlXJg2XNoAHADeACAAW-IAaMCkgEDMS4ymAEAoAEBoAECqgEHZ3dzLXdpesgBCMABAQ&sclient=psy-ab&ved=0ahUKEwiwwrO7mKbsAhVMrp4KHeTYCFEQ4dUDCA0&uact=5). I can change the signature to "ipv4_is_non_first_fragment" if it sounds better. But I don't want to get rid of the well understood name. WDYT

Reply:
Good point, poor suggestion on my part. How about something like `has_l4_ports()`? That's what really matters for whether we are performing the L4 checksumming.

Reply:
OK. added an ipv4_has_l4_header(). Don't like "port" because it may be icmp without ports.